frontend/src/pages/Main.tsx

```diff
@@ -284,9 +376,20 @@ const Main: FC = () => {
           onSelectedPath={(index, _road, position) => {
             if (roads === undefined) return;
 
+            // Set selected road index and position
+            setSelectedRoadIdx(index);
+            setMoveToPosition(position);
+
```

Comment:
why?

Reply:
I mean okay to move the `setMoveToPosition(position);` before the `if` but i don't understand why you are adding the `setSelectedRoadIdx(index);`

Reply:
Yea, I should remove this setSelectedRoadIdx(index);